test/gateway/eip20_gateway/revert_stake.js

```diff
@@ -52,9 +66,254 @@ contract('EIP20Gateway.revertStake()', function (accounts) {
+      {from: stakeMessage.staker},
+    );
+
+    let event = EventDecoder.getEvents(tx, gateway);
+
+    assert.isDefined(
+      event.RevertStakeIntentDeclared,
+      'Event `RevertStakeIntentDeclared` must be emitted.',
+    );
+
+    let eventData = event.RevertStakeIntentDeclared;
+    assert.strictEqual(
+      eventData._messageHash,
+      stakeMessage.messageHash,
+      `Expected message hash ${eventData._messageHash} is different from actual message hash ${stakeMessage.messageHash}`,
+    );
+
+    assert.strictEqual(
+      eventData._staker,
+      stakeMessage.staker,
+      `Expected message hash ${eventData._staker} is different from actual message hash ${stakeMessage.staker}`,
+    );
+
+    assert.strictEqual(
+      eventData._amount.eq(stakeRequest.stakeAmount),
+      true,
+      `Expected stake amount ${eventData._amount} is different from actual stake amount ${stakeRequest.stakeAmount}`,
+    );
+
+    assert.strictEqual(
+      eventData._stakerNonce.eq(stakeMessage.stakerNonce),
+      true,
+      `Expected staker nonce ${eventData._stakerNonce} is different from actual staker nonce ${stakeMessage.stakerNonce}`,
+    );
+
+  });
+
+  it('should charge/transfer penalty', async function () {
+
+    let gatewayInitialBalance = await baseToken.balanceOf(gateway.address);
+    let stakerInitialBalance = await baseToken.balanceOf(stakeMessage.staker);
+
+    let penalty = new BN(150);
+
+    await baseToken.approve(gateway.address, penalty, {from: stakeMessage.staker});
+    await gateway.setOutboxStatus(
+      stakeMessage.messageHash,
+      MessageStatusEnum.Declared,
+    );
+
+    await gateway.revertStake(
+      stakeMessage.messageHash,
+      {from: stakeMessage.staker},
+    );
+
+    let gatewayFinalBalance = await baseToken.balanceOf(gateway.address);
+    let stakerFinalBalance = await baseToken.balanceOf(stakeMessage.staker);
+
+    assert.strictEqual(
+      gatewayFinalBalance.eq(gatewayInitialBalance.add(penalty)),
+      true,
+      'Penalty must be transferred to gateway',
+    );
+
+    assert.strictEqual(
+      stakerFinalBalance.eq(stakerInitialBalance.sub(penalty)),
+      true,
+      'Penalty must be transferred from staker',
+    );
   });
 
-  it('should fail when messagehash is zero', async function () {
+  it('should fail for zero message hash', async function () {
+
+    let penalty = new BN(150);
+
+    await baseToken.approve(gateway.address, penalty, {from: stakeMessage.staker});
+    await gateway.setOutboxStatus(
+      stakeMessage.messageHash,
+      MessageStatusEnum.Declared,
+    );
+
+    await Utils.expectRevert(
+      gateway.revertStake(
+        Utils.ZERO_BYTES32,
+        {from: stakeMessage.staker},
+      ),
+      'Message hash must not be zero.'
+      );
+
+  });
+
+  it('should fail if revocation is already declared', async function () {
+
+    let penalty = new BN(150);
+
+    await baseToken.approve(gateway.address, penalty, {from: stakeMessage.staker});
```

Comment:
Following code is used in all test cases, this can be moved to a function.
```
    await baseToken.approve(gateway.address, penalty, {from: stakeMessage.staker});
    await gateway.setOutboxStatus(
      stakeMessage.messageHash,
      MessageStatusEnum.Declared,
    );
```

Reply:
Maybe if you find a good function name for the function that does these two things, then it is still readable and clear when you make the function call of that function in these places.

Reply:
I couldn't think of function name without `And` in it, do you have suggestion? 

Do you strongly feel this must be fixed?

Reply:
For writing, test cases don't have a strong view on what should we follow here. I am fine even if it is in this way.